src/utils/Plug.ts

```diff
@@ -0,0 +1,36 @@
+import { Lock } from '@matrixai/async-locks';
+
+/**
+ * Abstraction for using a Lock as a plug for asynchronous pausing of loops
+ */
+class Plug {
+  protected lock: Lock = new Lock();
+  protected lockReleaser: (e?: Error) => Promise<void> = async () => {};
```

Comment:
I don't think this abstraction is necessary. I'm pretty sure using the `Lock` is sufficiently obvious what it is doing.

Reply:
It's obvious yes, but I didn't want to replicate the logic every time I needed to use a lock as a plug. I can in-line it but I feel using a `plug` object has better readability.

Reply:
This has been removed